src/Core.jl

```diff
@@ -77,6 +77,85 @@ connector(::Type{Char}) = ""
 connector(::Type{Int}) = "∘"
 connector(::Type) = "-"
 
+function is_binary_tree(code::NestedEinsum)
+    if isleaf(code) return true end
+    if length(code.args) > 2 return false end
+    return all(is_binary_tree, code.args)
```

Comment:
This is correct and is necessary, since I found that KaHyPar sometimes return contraction tree with nodes having only 1 child. Correspondingly, the `pivot_tree` function is designed to handle both cases.

Reply:
Then it is not a binary tree. Maybe change the function name?

Reply:
I have changed it to `is_unary_or_binary`.